### DISCUSSION TITLE
addPassthroughCopy can't copy same source folder into several target directories

### DISCUSSION THREAD

**Discussion by alekspetrov** [11ty/eleventy#4038] [Enhancement Queue]
I'm trying to create multi-language webpage.
I need to copy `/assets` into every language folder:

**eleventy.js**
```
module.exports = function(eleventyConfig) {
  eleventyConfig.addPassthroughCopy({ assets: "en" });
  eleventyConfig.addPassthroughCopy({ assets: "fr" });
};
```

**Project structure:**
```
  _data
  _includes
  assets
  en
  fr
```

But `/assets` copied only into `/fr` language.

Any ideas?

**Comment by pdehaan** [comment]
Interesting. It looks like [`addPassthroughCopy()`](https://github.com/11ty/eleventy/blob/0d18f4d1ca4c0587c1ebba298a9603ca11429243/src/UserConfig.js#L299-L316) is proxying an internal [`passthroughCopies`](https://github.com/11ty/eleventy/blob/0d18f4d1ca4c0587c1ebba298a9603ca11429243/src/UserConfig.js#L44) property which is a simple object, so I'm guessing that using the same key ("assets") is just overwriting the existing value in the `passthroughCopies` object.

I wonder if you'd have to copy the assets yourself using a 3rd party lib like **copy**, or **cp**, or **fs-extra**.

https://github.com/11ty/eleventy/blob/0d18f4d1ca4c0587c1ebba298a9603ca11429243/src/UserConfig.js#L299-L316

https://github.com/11ty/eleventy/blob/0d18f4d1ca4c0587c1ebba298a9603ca11429243/src/UserConfig.js#L44

**Comment by pdehaan** [comment]
Not the prettiest solution, but I liked **fs-extra**'s API the best:

```js
const path = require("path");

const fs = require("fs-extra");

module.exports = eleventyConfig => {
  /*
  // The src/assets files are only copied to the "de/" (last) folder.
  eleventyConfig.addPassthroughCopy({ "src/assets": "en" });
  eleventyConfig.addPassthroughCopy({ "src/assets": "fr" });
  eleventyConfig.addPassthroughCopy({ "src/assets": "de" });
  */

  const dir = {
    input: "src",
    output: "www"
  };

  const locales = ["en", "fr", "de"];
  for (const locale of locales) {
    try {
      fs.copySync(
        // src: "./src/assets/"
        path.join(dir.input, "assets"),
        // dest: "./www/{{ locale }}/"
        path.join(dir.output, locale)
      );
    } catch (err) {
      console.error(err.message);
      process.exitCode = 1;
    }
  }

  return {
    dir
  };
};
```

### INPUT

```sh
$ tree src

src/
├── _data/
├── _includes/
├── assets/
│   ├── site.css
│   └── site.js
└── index.njk

3 directories, 3 files
```

### OUTPUT

```sh
$ tree www
www/
├── de/
│   ├── site.css
│   └── site.js
├── en/
│   ├── site.css
│   └── site.js
├── fr/
│   ├── site.css
│   └── site.js
└── index.html

3 directories, 7 files
```

**Comment by zachleat** [comment]
I’d imagine that syntax would probably be something like:

```js
eleventyConfig.addPassthroughCopy({ "src/assets": ["en", "de", "fr"] });
```

**Comment by zachleat** [comment]
> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by pdehaan** [comment]
> ```js
> eleventyConfig.addPassthroughCopy({ "src/assets": ["en", "de", "fr"] });
> ```

This seems to fail with the following message:

```
> Having trouble copying

`TemplateWriterWriteError` was thrown
> Having trouble copying './src/assets'

`TemplatePassthroughManagerCopyError` was thrown
> The "path" argument must be of type string. Received type object

`TypeError` was thrown:
    TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type object
```
